Packages/com.unity.multiplayer.samples.coop/Utilities/SceneManagement/SceneLoaderWrapper.cs

```diff
@@ -159,14 +159,21 @@ void OnSceneEvent(SceneEvent sceneEvent)
                     break;
                 case SceneEventType.Synchronize: // Server told client to start synchronizing scenes
                 {
-                    // todo: this is a workaround that could be removed once MTT-3363 is done
                     // Only executes on client that is not the host
                     if (NetworkManager.IsClient && !NetworkManager.IsHost)
                     {
-                        // unload all currently loaded additive scenes so that if we connect to a server with the same
-                        // main scene we properly load and synchronize all appropriate scenes without loading a scene
-                        // that is already loaded.
-                        UnloadAdditiveScenes();
+                        if (NetworkManager.SceneManager.ClientSynchronizationMode == LoadSceneMode.Single)
+                        {
+                            // If using the Single ClientSynchronizationMode, unload all currently loaded additive
+                            // scenes. In this case, we want the client to only keep the same scenes loaded as the
+                            // server. Netcode For GameObjects will automatically handle loading all the scenes that the
+                            // server has loaded to the client during the synchronization process. If the server's main
+                            // scene is different to the client's, it will start by loading that scene in single mode,
+                            // unloading every additively loaded scene in the process. However, if the server's main
+                            // scene is the same as the client's, it will not automatically unload additive scenes, so
+                            // we do it manually here.
```

Comment:
just a note: for multi-line comments, it's recommended to use `/* multiline comments */` instead of a series of `// `, as it reduces the visual noise

Reply:
In this case I kept it this way for consistency with the rest of the project, but perhaps we could create a task to update all our comments at once?